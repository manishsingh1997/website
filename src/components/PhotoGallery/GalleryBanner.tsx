import React from 'react';
import {NavLink} from 'react-router-dom';

type GalleryBannerProps = {
  imageSrc: string;
  header: string;
  description: string;
  url: string;
};

const GalleryBanner = ({imageSrc, header, description, url}: GalleryBannerProps) => {
  return (
    <div className="card soft-border photo-gallery__banner">
      <div className="photo-gallery__wrapper">
        <img className="photo-gallery__wrapper-photo" src={imageSrc} />
        <div className="photo-gallery__wrapper-data">
          <h4 className="additional-header h2 spacing after__is-6">{header}</h4>
          <p className="spacing after__is-12">{description}</p>
          <NavLink className="button button--regular taste__line button--size__medium is__button-desktop" to={url}>
            Open Gallery
          </NavLink>
        </div>
      </div>
      <NavLink className="button button--regular taste__line button--size__medium is__button-phone" to={url}>
        Open Gallery
      </NavLink>
    </div>
  );
};

export default GalleryBanner;
