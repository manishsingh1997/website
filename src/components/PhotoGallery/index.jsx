import React from 'react';
import PropTypes from 'prop-types';
import find from 'lodash/find';
import Carousel, {Modal, ModalGateway} from 'react-images';
import Masonry from 'react-masonry-component';
import {NavLink} from 'react-router-dom';
import isEmpty from 'lodash/isEmpty';
import isNil from 'lodash/isNil';

import {Tabs, Breadcrumb, ImageCard} from '@ergeon/core-components';

import {
  FencePhotoData,
  GatePhotoData,
  DrivewayPhotoData,
} from 'data/photo-gallery';

const PRODUCTS = {
  FENCE: 'fence',
  GATE: 'gate',
  DRIVEWAY: 'driveway',
};

const BANNERS = {
  FENCE_BANNER: {
    imageSrc: 'https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/previews/banner_fence.jpg',
    header: 'Fence Photos',
    description: 'Picture Frame, Nail Up, Horizontal, Gates, Before & After',
    url: '/gallery/fence/picture-frame',
  },
  GATE_BANNER: {
    imageSrc: 'https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/previews/banner_gate.jpg',
    header: 'Gate Photos',
    description: 'Picture Frame, Nail Up, Horizontal, Gates, Before & After',
    url: '/gallery/gate/single',
  },
  DRIVEWAY_BANNER: {
    imageSrc: 'https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/previews/banner_driveway.jpg',
    header: 'Driveway Photos',
    description: 'Stamped & Stained Concrete, Brushed Concrete, Pavers',
    url: '/gallery/driveway/stamped',
  },
};

import './index.scss';

class PhotoGallery extends React.Component {
    static propTypes = {
      children: PropTypes.node,
      history: PropTypes.object,
      location: PropTypes.object,
      match: PropTypes.shape({
        params: PropTypes.shape({
          categorySlug: PropTypes.string,
          groupSlug: PropTypes.string,
          productSlug: PropTypes.oneOf(Object.values(PRODUCTS)),
        }),
      }),
    };

    state = {
      modalOpened: false,
      imageIndex: 0,
    };

    updateLocationHash(slug) {
      const {history, location: {pathname, search}} = this.props;
      history.push(`${pathname}${search}#${slug}`);
    }

    handlePhotoClick(photoProps, index) {
      this.updateLocationHash(photoProps.slug);
      this.setState({modalOpened: true, imageIndex: index});
    }

    clearLocationHash() {
      const {history, location: {pathname, search}} = this.props;
      history.push(`${pathname}${search}`);
    }

    handleModalClose() {
      this.clearLocationHash();
      this.setState({modalOpened: false});
    }

    getProductData() {
      const {productSlug} = this.props.match.params;
      const dataByProduct = {
        [PRODUCTS.FENCE]: FencePhotoData,
        [PRODUCTS.GATE]: GatePhotoData,
        [PRODUCTS.DRIVEWAY]: DrivewayPhotoData,
      };

      return dataByProduct[productSlug] || dataByProduct[PRODUCTS.FENCE];
    }

    getCategoryData() {
      const {categorySlug} = this.props.match.params;
      const product = this.getProductData();
      return find(product, {categorySlug}) || {};
    }

    getBreadcrumbGroups() {
      const {productSlug, categorySlug, groupSlug} = this.props.match.params;
      const category = this.getCategoryData();
      const categoryGroups = category.categoryGroups || [];

      return categoryGroups.map(group => ({
        name: group.groupName,
        to: `/gallery/${productSlug}/${categorySlug}/${group.groupSlug}`,
        highlighted: groupSlug === group.groupSlug,
      }));
    }

    getCategoryTabs() {
      const {productSlug, categorySlug} = this.props.match.params;

      return this.getProductData().map(category => ({
        content: category.categoryName,
        id: `tab-${category.categorySlug}`,
        to: `/gallery/${productSlug}/${category.categorySlug}`,
        active: categorySlug === category.categorySlug,
        exact: false,
      }));
    }

    getProductGalleryName() {
      const {productSlug} = this.props.match.params;
      const productGalleryNames = {
        [PRODUCTS.FENCE]: 'Fence Photos',
        [PRODUCTS.GATE]: 'Gate Photos',
        [PRODUCTS.DRIVEWAY]: 'Driveway Photos',
      };

      return productGalleryNames[productSlug || 'fence'] || productGalleryNames['fence'];
    }

    getPhotos() {
      const {groupSlug} = this.props.match.params;
      const category = this.getCategoryData();
      const group = find(category.categoryGroups, {groupSlug});

      if (groupSlug && group) return group.groupPhotos.map((elem) => {
        // there's some groupPhotos that doesn't contain caption, so lets map title to them
        if (!elem.caption && elem.title) return {caption: elem.title, ...elem};
        return elem;
      }) || [];
      return category.categoryPhotos || [];
    }

    layoutImages() {
      if (this.masonry) {
        setTimeout(() => {
          this.masonry.layout();
        });
      }
    }

    renderPhotos() {
      const photos = this.getPhotos();

      return (
        <Masonry
          className="photo-gallery__masonry"
          ref={ref => this.masonry = this.masonry || ref?.masonry}>
          {!isEmpty(photos) && photos.map((photoProps, index) => (
            <div
              className="photo-gallery__masonry-card"
              gallery-link-id={photoProps.slug}
              key={photoProps.slug}
              onClick={this.handlePhotoClick.bind(this, photoProps, index)}>
              <ImageCard onImageLoaded={this.layoutImages.bind(this)} {...photoProps} />
            </div>
          ))}
        </Masonry>
      );
    }

    renderBanner({imageSrc, header, description, url}, key) {
      return (
        <div className="card soft-border photo-gallery__banner" key={key}>
          <img className="photo-gallery__banner-photo" src={imageSrc} />
          <div className="photo-gallery__banner-data">
            <h4 className="additional-header h2 spacing after__is-6">{header}</h4>
            <p className="spacing after__is-12">
              {description}
            </p>
            <NavLink
              className="button button--regular taste__line button--size__medium"
              to={url}>
              Open Gallery
            </NavLink>
          </div>
        </div>
      );
    }

    renderPartners(partners = []) {
      if (isNil(partners)) return;
      return (
        <div className="photo-gallery__partners-area">
          <label className="label uppercase spacing after__is-12">
            FOR ADDITIONAL CUSTOMIZED OPTIONS VISIT OUR PARTNER LINKS
          </label>
          <div className="photo-gallery__partners">
            {partners.map(({name, slug, url, imageSrc}) => (
              <a
                className="photo-gallery__partner card soft-border shadow__z1"
                href={url}
                key={`partner-${slug}`}
                rel="noopener noreferrer"
                target="_blank">
                <div className="photo-gallery__partner-image" style={{backgroundImage: `url(${imageSrc})`}} />
                <div className="photo-gallery__partner-description">
                  <div>
                    <h4 className="additional-header h4">
                      {name}
                    </h4>
                    <div className="photo-gallery__partner-site">
                      {url}
                    </div>
                  </div>
                </div>
              </a>
            ))}
          </div>
        </div>
      );
    }

    render() {
      const {productSlug} = this.props.match.params;
      if (!productSlug) {
        return (
          <div className="photo-gallery">
            <div className="wrapper-1180">
              <span className="breadcrumbs">
                <NavLink to="/"><span className="icon home"/></NavLink>
                <ul>
                  <li><NavLink to="/gallery/">Photo gallery</NavLink></li>
                </ul>
              </span>
              <h1 className="h2">Photo gallery</h1>
            </div>
            <div className="header-border spacing after__is-24">
            </div>
            <div className="wrapper-1180">
              <div className="cards two-columns">
                {[BANNERS.GATE_BANNER, BANNERS.DRIVEWAY_BANNER, BANNERS.FENCE_BANNER].map(
                  (banner, index) => this.renderBanner(banner, index)
                )}
              </div>
            </div>
          </div>
        );
      }

      const {modalOpened, imageIndex} = this.state;
      const category = this.getCategoryData();
      const breadcrumbGroups = this.getBreadcrumbGroups();
      const activeGroup = find(breadcrumbGroups, {highlighted: true});

      const banners = {
        [PRODUCTS.FENCE]: [BANNERS.GATE_BANNER, BANNERS.DRIVEWAY_BANNER],
        [PRODUCTS.GATE]: [BANNERS.FENCE_BANNER, BANNERS.DRIVEWAY_BANNER],
        [PRODUCTS.DRIVEWAY]: [BANNERS.FENCE_BANNER, BANNERS.GATE_BANNER],
      }[productSlug];

      const photos = this.getPhotos();

      return (
        <div className="photo-gallery">
          <div className="wrapper-1180">
            <span className="breadcrumbs">
              <NavLink to="/"><span className="icon home"/></NavLink>
              <ul>
                <li><NavLink to="/gallery/">Photo gallery</NavLink></li>
              </ul>
            </span>
            <h1 className="h2">{category?.categoryName} {activeGroup?.name} {this.getProductGalleryName()}</h1>
          </div>
          <div className="header-border">
            <div className="photo-gallery__categories wrapper-1180">
              <Tabs items={this.getCategoryTabs()} useNavLinks />
            </div>
          </div>
          <div className="wrapper-1180">
            {breadcrumbGroups.length > 1 && <div className="photo-gallery__groups spacing after__is-24">
              <Breadcrumb
                items={[breadcrumbGroups]} secondary/>
            </div>}
            {this.renderPhotos()}
            <div className="cards two-columns spacing before__is-64">
              {banners.map((banner, index) => this.renderBanner(banner, index))}
            </div>
            {category.partners && this.renderPartners(null)}
            <ModalGateway>
              {!isEmpty(photos) && modalOpened ? (
                <Modal onClose={() => this.handleModalClose()}>
                  <Carousel
                    currentIndex={imageIndex}
                    trackProps={{
                      onViewChange: index => this.updateLocationHash(photos[index].slug),
                    }}
                    views={photos.map(({caption, url}) => ({caption, src: url}))}/>
                </Modal>
              ) : null}
            </ModalGateway>
          </div>
        </div>
      );
    }
}

export default PhotoGallery;
