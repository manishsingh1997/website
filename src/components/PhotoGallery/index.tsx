import React from 'react';

// eslint-disable-next-line import/named
import {match, RouteComponentProps} from 'react-router-dom';

import './index.scss';

import GalleryPreview from './GalleryPreview';
import GalleryContent from './GalleryContent';

type Params = {
  categorySlug: string;
  groupSlug: string;
  productSlug: string;
};

export type PhotoGalleryProps = {
  children?: React.ReactNode;
  match: match<Params>;
} & RouteComponentProps;

const PhotoGallery = ({history, location, match}: PhotoGalleryProps) => {
  const {productSlug} = match.params;

  if (!productSlug) {
    return <GalleryPreview />;
  }

  return <GalleryContent history={history} location={location} match={match} />;
};

export default PhotoGallery;
