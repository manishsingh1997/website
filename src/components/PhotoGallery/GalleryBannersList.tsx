import React from 'react';

import { GALLERY_BANNERS } from '../../data/gallery-banners-data';

import GalleryBanner from './GalleryBanner';

type GalleryBannersListProps = {
  productSlug: string;
  products: { [key: string]: string };
};

const GalleryBannersList = ({ productSlug, products }: GalleryBannersListProps) => {
  const banners = {
    [products.FENCE]: [GALLERY_BANNERS.GATE_BANNER],
    [products.GATE]: [GALLERY_BANNERS.FENCE_BANNER],
    [products.GRASS]: [GALLERY_BANNERS.GRASS_BANNER],
  }[productSlug];

  return (
    <div className="cards two-columns spacing before__is-64">
      {banners.map((banner, index) => {
        if (banner.header === 'Artificial Grass') return null;
        return (
          <GalleryBanner key={index} {...banner} />
        )
      })}
    </div>
  );
};

export default GalleryBannersList;
