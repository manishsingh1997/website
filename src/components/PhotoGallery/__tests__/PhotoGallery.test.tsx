import React from 'react';

import '@testing-library/jest-dom';
import {createMemoryHistory, createLocation} from 'history';
import {render, screen} from '@testing-library/react';

import PhotoGallery from '../';
import {GALLERY_BANNERS} from '../../../data/gallery-banners-data';

const mockProps = {
  location: createLocation(''),
  history: createMemoryHistory(),
  match: {
    isExact: true,
    path: '/gallery',
    url: '/gallery/',
    params: {
      productSlug: '',
      categorySlug: '',
      groupSlug: '',
    },
  },
};

jest.mock('react-router-dom', () => ({
  Link: jest.fn(({to}) => `Link to ${to}`),
  NavLink: jest.fn(({to}) => `NavLink to ${to}`),
}));

describe('Photo gallery page without params', () => {
  beforeEach(() => {
    render(<PhotoGallery {...mockProps} />);
  });
  it('should show the title correctly', () => {
    const pageHeading = screen.getByRole('heading', {name: 'Photo gallery'});
    expect(pageHeading).toBeInTheDocument();
  });
  it('should render the correctly banners length', () => {
    const img = screen.getAllByRole('img');
    expect(img).toHaveLength(Object.keys(GALLERY_BANNERS).length);
  });
});

describe('Photo gallery page with params', () => {
  it('should match with snapshot', () => {
    const mockWithParams = {
      ...mockProps,
      match: {
        ...mockProps.match,
        params: {
          productSlug: 'fence',
          categorySlug: 'picture-frame',
          groupSlug: 'without-lattice',
        },
      },
    };

    const {container} = render(<PhotoGallery {...mockWithParams} />);
    expect(container).toMatchSnapshot();
  });
});
