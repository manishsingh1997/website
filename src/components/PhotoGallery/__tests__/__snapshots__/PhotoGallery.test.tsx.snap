// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo gallery page with params should match with snapshot 1`] = `
<div>
  <div
    class="photo-gallery"
  >
    <div
      class="wrapper-1180"
    >
      <span
        class="breadcrumbs"
      >
        NavLink to /
        <ul>
          <li>
            NavLink to /gallery/
          </li>
        </ul>
      </span>
      <h1
        class="h2"
      >
        Picture Frame
         
        Without Lattice
         
        Fence Photos
      </h1>
    </div>
    <div
      class="header-border"
    >
      <div
        class="photo-gallery__categories wrapper-1180"
      >
        <ul
          class="tabs tabs--with-nav-links"
        >
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/picture-frame
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/nail-up
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/horizontal
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/chain-link
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/boxed-frame
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/staining
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/lumber
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/before-after
          </li>
          <li
            class="navlink"
          >
            NavLink to /gallery/fence/vinyl
          </li>
        </ul>
      </div>
    </div>
    <div
      class="wrapper-1180"
    >
      <div
        class="photo-gallery__groups spacing after__is-24"
      >
        <div
          class="breadcrumb"
        >
          <div
            class="breadcrumb--group"
          >
            NavLink to /gallery/fence/picture-frame/without-lattice
            NavLink to /gallery/fence/picture-frame/with-lattice
            NavLink to /gallery/fence/picture-frame/short-fence
          </div>
        </div>
      </div>
      <div
        class="photo-gallery__masonry"
        style="position: relative; height: 0px;"
      >
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="picture-frame-side-by-side-stepped-top"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Picture Frame Side by Side Stepped Top"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/pf+sbs+2.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Picture Frame Side by Side Stepped Top
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="picture-frame-side-by-side"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Picture Frame Side by Side"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/PF+sbs.jpeg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Picture Frame Side by Side
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="picture-frame-louvered-with-kickboard-1"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Picture Frame Louvered with Kickboard"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/pf+louvered.png"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Picture Frame Louvered with Kickboard
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="picture-frame-louvered-with-kickboard-2"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Picture Frame Louvered with Kickboard"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/pf+louvered+3.png"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Picture Frame Louvered with Kickboard
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="vallejo-ca-picture-frame-flat-top-with-kickboard"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Vallejo, CA - Picture Frame Board on Board with Kickboard"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5cc086dd9e1f671c728112fe_vallejo%2520picture%2520frame.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Vallejo, CA - Picture Frame Board on Board with Kickboard
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="palo-alto-ca-picture-frame-board-on-board-with-an-integrated-3-retaining-wall"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Palo Alto, CA - Picture Frame Board on Board with an integrated 3' Retaining Wall"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5cc085e10a5c75c28bbabb01_paloalto%2520pictureframe%2520stepped.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Palo Alto, CA - Picture Frame Board on Board with an integrated 3' Retaining Wall
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="pleasanton-ca-picture-frame-board-on-board-flat-top-on-concrete-retaining-wall"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Pleasanton, CA - Picture Frame Board on Board, Flat Top on Concrete Retaining Wall"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c61e57becd0150b0107494c_picture%2520frame%2520concrete%2520retaining%2520wall.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Pleasanton, CA - Picture Frame Board on Board, Flat Top on Concrete Retaining Wall
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="berkeley-ca-picture-frame-side-by-side-with-1-gap-flat-top"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Berkeley, CA - Picture Frame Side by Side with 1\\" gap, Flat Top"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c5e07a7364f641708f5375c_picture%2520frame%2520flat%2520top%2520side%2520gap%2520between%2520pickets.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Berkeley, CA - Picture Frame Side by Side with 1" gap, Flat Top
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="american-canyon-ca-picture-frame-board-on-board-flat-top-with-size-trim"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="American Canyon, CA - Picture Frame Board on Board, Flat Top with Side Trim"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c5e068532efce2920608182_picture%2520frame%2520flat%2520top%2520side%2520trim.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              American Canyon, CA - Picture Frame Board on Board, Flat Top with Side Trim
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="sacramento-ca-picture-frame-board-on-board-flat-top"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Sacramento, CA - Picture Frame Board on Board, Flat Top "
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c5e05712f64c67364650967_picture%2520frame%2520flat%2520top%25203.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Sacramento, CA - Picture Frame Board on Board, Flat Top 
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="albany-ca-picture-frame-board-on-board-with-flat-top"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Albany, CA - Picture Frame Board on Board, Flat Top "
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c5e03d18d94f53b1769ea82_picture%2520frame%2520flat%2520top.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Albany, CA - Picture Frame Board on Board, Flat Top 
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="woodside-ca-picture-frame-board-on-board-with-an-integrated-3-retaining-wall"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Woodside, CA - Picture Frame Board on Board with an integrated 3' Retaining Wall"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c2530874fdbbaf99bbf2625_retaining%2520wall%2520fence6.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Woodside, CA - Picture Frame Board on Board with an integrated 3' Retaining Wall
            </p>
          </div>
        </div>
        <div
          class="photo-gallery__masonry-card"
          gallery-link-id="morgan-hill-ca-picture-frame-with-1-x1-s-top-and-bottom-board-on-board-with-an-integrated-1-retaining-wall"
          style="position: absolute; left: 0px;"
        >
          <div
            class="card padding-20 soft-border shadow__z1"
          >
            <div
              class="image-loader__wrapper is-loading"
            >
              <div
                class="image-loader__spinner"
              >
                <div
                  class="loader-wrapper"
                  style="width: 64px; height: 64px;"
                >
                  <div
                    class="lightgray loader"
                  >
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                    <div
                      style="border-width: 10px; width: 64px; height: 64px;"
                    />
                  </div>
                </div>
              </div>
              <img
                alt="Morgan Hill, CA - Picture Frame Board on Board with an integrated 1' Retaining Wall"
                class="image-loader__image"
                src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/fence/5c2530164fdbba8288bf2532_retaining%2520wall%2520fence5.jpg"
              />
            </div>
            <div
              class="additional-header h2 spacing before__is-6 after__is-6"
            />
            <p />
            <p>
              Morgan Hill, CA - Picture Frame Board on Board with an integrated 1' Retaining Wall
            </p>
          </div>
        </div>
      </div>
      <div
        class="cards two-columns spacing before__is-64"
      >
        <div
          class="card soft-border photo-gallery__banner"
        >
          <div
            class="photo-gallery__wrapper"
          >
            <img
              class="photo-gallery__wrapper-photo"
              src="https://ergeon-photo-gallery.s3-us-west-1.amazonaws.com/previews/banner_gate.jpg"
            />
            <div
              class="photo-gallery__wrapper-data"
            >
              <h4
                class="additional-header h2 spacing after__is-6"
              >
                Gate Photos
              </h4>
              <p
                class="spacing after__is-12"
              >
                Picture Frame, Nail Up, Horizontal, Gates, Before & After
              </p>
              NavLink to /gallery/gate/single
            </div>
          </div>
          NavLink to /gallery/gate/single
        </div>
      </div>
    </div>
  </div>
</div>
`;
